pkg/sync/queue_manager.go

```diff
@@ -160,6 +160,35 @@ func getQueueKey(run *tektonv1.PipelineRun) string {
 	return fmt.Sprintf("%s/%s", run.Namespace, run.Name)
 }
 
+// FilterPipelineRunByInProgress filters the given list of PipelineRun names to only include those
+// that are in a "queued" state and have a pending status. It retrieves the PipelineRun objects
+// from the Tekton API and checks their annotations and status to determine if they should be included.
+//
+// Returns A list of PipelineRun names that are in a "queued" state and have a pending status.
+func FilterPipelineRunByInProgress(ctx context.Context, tekton versioned2.Interface, orderList []string, wantedStatus, wantedState string) []string {
```

Comment:
Shouldn't the Function `FilterPipelineRunByState` because it has wantedState param so I guess it's not specific for InProgress PipelineRuns but it's generic for all states.

Reply:
sounds good will rename